authn/pkce.index.js

```diff
@@ -362,9 +362,9 @@ function unauthorized(error, error_description, error_uri, callback) {
   </html>
   `;
 
-  page = page.replace(/%error%/g, error);
-  page = page.replace(/%error_description%/g, error_description);
-  page = page.replace(/%error_uri%/g, error_uri);
+  page = page.replace(/%error%/g, encodeURI(error).replace(/%20/g,' '));
+  page = page.replace(/%error_description%/g, encodeURI(error_description).replace(/%20/g,' '));
```

Comment:
This may make the supplied string "safe" but I don't think URI encoding is what e are going to want. encodeURI is designed to encode urls not strings shown as HTML. I think we will need some form of HTML Encoding instead.

For example when encoding the string `<script>alert('boo & hiss');</script>` the **encodeURI** results in `%3Cscript%3Ealert('boo%20&%20hiss');%3C/script%3E`.
Where as I think we would want it to be displayed as `&lt;script&gt;alert('boo &amp; hiss');&lt;/script&gt;`

I don't think there is a native function for HTML encoding in JavaScript, but some searching will confirm or deny a soln there.

Reply:
I think you're right, HTML encoding will be better than URI encoding for this text.
I'll look at moving it over, and probably also update it in the [openid.index.js](https://github.com/iress/cloudfront-auth/blob/f4c0e4491b52a93614ff2ce1add76832a5c48218/authn/openid.index.js#L338C18-L338C18) file as well to align them.

Reply:
I've installed and used the [entities npm package](https://www.npmjs.com/package/entities) to do HTML encoding instead of URI for both files.

Reply:
@SiCoe quick follow up question... have you checked adding the additional package does not break the size limit for lambda@edge functions as mentioned in the [readme](https://github.com/iress/cloudfront-auth#contributing)?

Reply:
the `entites` package looks good from a [health pov](https://snyk.io/advisor/npm-package/entities) but I notice it has a _BSD-2-Clause_ licence. Is that licence compatible with this repo?

Reply:
Looks like the size of the package listed in the action results is under the required 1MB
![Screenshot 2023-09-26 at 11 45 24](https://github.com/iress/cloudfront-auth/assets/48359419/7c508cb7-991a-4a51-b1dc-6db0449c5c4f)

Reply:
> the `entites` package looks good from a [health pov](https://snyk.io/advisor/npm-package/entities) but I notice it has a _BSD-2-Clause_ licence. Is that licence compatible with this repo?

@kristens-work do you know, or know anyone who's able to advise on use of a BSD 2 Clause licenced dependencies within our open-source repositories?

Reply:
I tend to seek advice from Rebecca Kelly on opensource licencing

Reply:
> the `entites` package looks good from a [health pov](https://snyk.io/advisor/npm-package/entities) but I notice it has a _BSD-2-Clause_ licence. Is that licence compatible with this repo?

I've replaced the dependency on the _BSD-2-Clause_ **[entities](https://www.npmjs.com/package/entities)** package with a dependency on the _MIT_ **[html-entities](https://www.npmjs.com/package/html-entities)** package.

Reply:
I'm happy that resolves the licensing question to no detriment to the code base.